packages/headless/src/controllers/core/did-you-mean/headless-core-did-you-mean.ts

```diff
@@ -31,6 +37,19 @@ export interface DidYouMeanProps {
    * The default value is `true`.
    */
   automaticallyCorrectQuery?: boolean;
+  // TODO: V3: Change the default to true
+  /**
+   * Whether to use machine learning powered query suggestions model as a fallback to provide query corrections.
+   * This system requires a working and properly configured query suggestions model in the Coveo platform.
+   *
+   * This option is off by default. As such, the Coveo platform will use an older query correction system, powered solely by the index.
+   * By opting in this new system, the Coveo Search API will stop returning the `queryCorrections` field in the response.
+   * Instead, it will start returning a `changedQuery` field.
+   * This implies that the usage of this option introduce a breaking change in the way query corrections are handled, both at the Search API and Headless level.
+   *
+   * The default value is `false`.
+   */
+  enableFallbackSearchOnEmptyQueryResults?: boolean;
```

Comment:
Should we name this `automaticallyCorrectQueryServerSide` or something along those lines? 

These two parameters are kind of dependant and could have been replaced by an enum:

```typescript
type automaticallyCorrectQuery: 'disabled' | 'client-side' | 'server-side'
```

Reply:
The type automaticallyCorrectQuery above (boolean) is already part of public Headless API (through global state/actions to control it).

And so we can't change it now to become an enum.

We could mark it as deprecated, however, and create a new one called automaticQueryCorrection: 'disabled' | 'client-side' | 'server-side'.

I kinda like that 🤔

Reply:
I like the enum too

Reply:
- [ ] enum